gnuplot/src/color.rs

```diff
@@ -0,0 +1,413 @@
+			VariableRGBInteger(_) => "rgb variable",
+			VariableARGBInteger(_) => "rgb variable",
+			PaletteFracColor(v) => &format!("palette frac {v}"),
+			PaletteCBColor(v) => &format!("palette cb {v}"),
+			VariablePaletteColor(_) => "palette z",
+			SavedColorMap(s, _) => &format!("palette {s}"),
+			VariableIndex(_) => "variable",
+			Background => "background",
+			Index(n) => &format!("{}", n),
+			Black => "black",
+		};
+		String::from(str)
+	}
+
+	pub fn data(&self) -> Vec<f64>
+	{
+		match self
+		{
+			RGBString(_) => panic!("data() called on non-variable color type."),
+			RGBInteger(_, _, _) => panic!("data() called on non-variable color type."),
+			ARGBInteger(_, _, _, _) => panic!("data() called on non-variable color type."),
+			VariableRGBInteger(items) => items
+				.iter()
+				.map(|(r, g, b)| from_argb(0, *r, *g, *b) as f64)
+				.collect(),
+			VariableARGBInteger(items) => items
+				.iter()
+				.map(|(a, r, g, b)| from_argb(*a, *r, *g, *b) as f64)
+				.collect(),
+			PaletteFracColor(_) => panic!("data() called on non-variable color type."),
+			PaletteCBColor(_) => panic!("data() called on non-variable color type."),
+			VariablePaletteColor(items) => items.clone(),
+			SavedColorMap(_, items) => items.clone(),
+			Index(_) => panic!("data() called on non-variable color type."),
+			VariableIndex(items) => items.iter().map(|v| *v as f64).collect(),
+			Background => panic!("data() called on non-variable color type."),
+			Black => panic!("data() called on non-variable color type."),
+		}
+	}
+
+	pub fn is_variable(&self) -> bool
+	{
+		matches!(
+			self,
+			VariableRGBInteger(_)
+				| VariableARGBInteger(_)
+				| VariableIndex(_)
+				| VariablePaletteColor(_)
+				| SavedColorMap(_, _)
+		)
+	}
+
+	pub fn has_alpha(&self) -> bool
+	{
+		match self
+		{
+			RGBString(s) =>
+			{
+				let s = s.to_string();
+				s.starts_with("0x") && s.chars().count() == 10
+					|| s.starts_with("#") && s.chars().count() == 9
+			}
+			ARGBInteger(_, _, _, _) | VariableARGBInteger(_) => true,
+			_ => false,
+		}
+	}
+}
+
+fn from_argb(a: ColorComponent, r: ColorComponent, g: ColorComponent, b: ColorComponent)
+	-> ColorInt
+{
+	((a as ColorInt) << 24) + ((r as ColorInt) << 16) + ((g as ColorInt) << 8) + (b as ColorInt)
+}
+
+fn float_color_to_int(v: f64) -> u8
+{
+	if !(0.0..=1.0).contains(&v)
+	{
+		panic!(
+			"Float value must be greater than zero and less than one. Actual value: {}",
+			v
+		);
+	}
+	((v * 255.0).round()) as u8
+}
+
+/// Converts a set of `f64` red, green and blue values in the range `0 <= x <= 1` to a 3-tuple of `u8` suitable for use as
+/// an [RGBInteger]
+///
+/// Panics if any of the arguments are not in the range `0 <= x <= 1`
+///
+/// Ses also [floats_to_argb]
+///
+/// # Arguments
+/// * r - red. 0: no red, 1: fully red
+/// * g - green. 0: no green, 1: fully green
+/// * b - blue. 0: no blue, 1: fully blue
+pub fn floats_to_rgb(r: f64, g: f64, b: f64) -> RGBInts
```

Comment:
What is the motivation behind this function? It doesn't seem like it adds enough functionality to justify increasing the size of the API (same for the argb variant).

Reply:
Yes, I agree: worth removing from the public API. The `TryInto` traits on tuples are more useful versions of these do.